### PR TITLE
Dataclasses support

### PR DESCRIPTION
**Pull request by TuanAnh17N**
**What this PR does / why we need it**:
This PR introduces support for handling dataclasses in the pipeline-definition-contract documentation generation. It refactors the current approach to ensure that dataclasses are properly processed recursively, aligning with the long-term goal of moving away from the "ModelBase" structure.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

**Comment by gardener-robot-ci-3** [comment]
Thank you @TuanAnh17N for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

**Comment by ccwienk** [comment]
@TuanAnh17N : you might ignore the failed github-action (or: rebase your pullrequest)

**Comment by zkdev** [comment]
`824b81fe02a4af5f2324eb604240158076118118` should not be part of this PR, please rebase against master

File: doc/lib/attributes.py [hunk 1]
```diff
@@ -64,7 +64,9 @@ def children(self):
         return self._child_elements
 
     def _type_name(self, type_):
-        if _is_generic_alias(type_):
+        if dataclasses.is_dataclass(type_):
+            return f'Dataclass[{type_.__name__}]'
```
Comment (ccwienk) [inline doc/lib/attributes.py:68]: just return the typename

**Review by ccwienk — APPROVED** [review]

**Pull request merged**